File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rsp/filterCodes/api/FilterCodesImpl.java
package org.ngbp.jsonrpc4jtestharness.rsp.filterCodes.api;

import org.ngbp.jsonrpc4jtestharness.models.JsonRpcResponse;
import org.ngbp.jsonrpc4jtestharness.rsp.filterCodes.model.GetFilterCodes;

import java.util.ArrayList;

public class FilterCodesImpl implements FilterCodes {
    @Override
    public GetFilterCodes getFilterCodes() {
        GetFilterCodes.Filters filters =  new GetFilterCodes.Filters();
        filters.expires =" 10";
        filters.filterCode = 10;
        ArrayList<GetFilterCodes.Filters> filtersList = new ArrayList<>();
        filtersList.add(filters);
        GetFilterCodes getFilterCodes =  new GetFilterCodes();
        getFilterCodes.filters = filtersList;
        return getFilterCodes;
    }

    @Override
    public JsonRpcResponse<Object> setFilterCodes() {
        return null;
    }
}

File: nextgen-middleware/src/main/java/com/nextgenbroadcast/mobile/middleware/rpc/filterCodes/FilterCodesImpl.kt
package com.nextgenbroadcast.mobile.middleware.rpc.filterCodes

import com.nextgenbroadcast.mobile.middleware.rpc.RpcResponse
import com.nextgenbroadcast.mobile.middleware.rpc.filterCodes.model.FilterCodes

class FilterCodesImpl : IFilterCodes {
    override fun getFilterCodes(): FilterCodes {
        val filters = FilterCodes.Filter(10, " 10")
        return FilterCodes(listOf(filters))
    }

    override fun setFilterCodes(filters: List<FilterCodes.Filter>): RpcResponse {
        return RpcResponse()
    }
}